Frameworks/Foundation/NSString.mm

```diff
@@ -2021,7 +2021,13 @@ - (id)initWithCoder:(NSCoder*)coder {
  @Status Interoperable
 */
 - (void)encodeWithCoder:(NSCoder*)coder {
-    [coder encodeObject:self forKey:@"NS.string"];
+    if ([coder isKindOfClass:[NSKeyedArchiver class]]) {
```

Comment:
That is to say: NSKeyedArchiver will encode a string that refers to itself as its string.